cvmfs/catalog_mgr_rw.cc

```diff
@@ -837,6 +847,34 @@ shash::Any WritableCatalogManager::SnapshotCatalog(WritableCatalog *catalog)
   return hash_catalog;
 }
 
+void WritableCatalogManager::Balance() {
+  CatalogList catalog_list = GetCatalogs();
```

Comment:
You might use it in a correct way but it is a public method and this class is generally thread-safe.  So if you don't lock, there should be a good reason (e.g. locking has been done by the caller), which should be documented.